dev-support/checkcompatibility.py

```diff
@@ -241,6 +241,20 @@ def compare_results(tool_results, known_issues, compare_warnings):
 
     return bool(unexpected_issues)
 
+def compare_tool_results_count(tool_results, check, issue_type, known_count):
+    """ Check problem counts are no more than the known count.
+    (This function exists just so can add in logging; previous was inlined
+    one-liner but this made it hard debugging)
+    """
+    # logging.info("known_count=%s, check key=%s, tool_results=%s, issue_type=%s",
+    #        str(known_count), str(check), str(tool_results), str(issue_type))
+    try:
+       return tool_results[check][issue_type] > known_count
+    except KeyError as ke:
+      logging.info("KeyError %s", str(ke))
```

Comment:
why log + swallow the error case here? Isn't it better to `raise` the error instead of just logging it? Or maybe calling code ignores thrown exceptions?

Reply:
Let me remove this bit... This was a debug method trying to see what was on other side of exception.... Let me make it fail.